main.nf

```diff
@@ -254,6 +261,27 @@ process generate_decoy_database {
 }
 
 
+/*
+ * STEP 1.5 - Optional: Run Peak Picking as Preprocessing
+ */
+process peak_picking {
+
+    input:
+     file mzml_unpicked from input_mzmls_unpicked
+
+    output:
+     file "${mzml_unpicked.baseName}.mzML" (into input_mzmls, input_mzmls_align)
```

Comment:
Here is another problem:

So I would like to introduce this optional preprocessing step. If the flag centroided == 'False' is set.

Now im creating the same channel in a different part of the script and that causes a problem:
`ERROR ~ Channel `input_mzmls` has been used twice as an output by process `peak_picking` and another operator
`

Reply:
This here is your friend :-) http://nextflow-io.github.io/patterns/index.html#_conditional_process_executions

Reply:
use `when` in combination with multiple channels and the `mix` operator

Reply:
ok I see, yes the mix operator seems to be the right choice for this